src/3.flatten/index.ts

```diff
@@ -5,7 +5,9 @@
 // No need to use recursion
 
 // ------- start coding here -------
-
+type Transformer<T extends { [P in any]: any }> = {
```

Comment:
`{ [P in any]: any }` ini selalu bisa dirubah ke `Record<any, any>` seperti yang diajarkan di materi Record di kelas.

Nah berhubung kita tidak menggunakan symbol (cuma string dan number) jadinya kita menggunakan `Record<string, any>`.

Kenapa kita tidak tulis `number | string`? Seperti yang diajarkan di kelas, ketika kita menulis string maka tipe data dari keynya sudah termasuk number juga (string | number).

Kenapa kita tidak menggunakan any? Karena any akan menghasilkan tipe data `string | number | symbol`, sedangkan kita tidak butuh symbol.

Untuk kedepannya gunakan `Record<string, any>`